music/src/core/player.ts

```diff
@@ -39,21 +39,76 @@ function compareQuantizedNotes(a: NoteSequence.INote, b: NoteSequence.INote) {
   return 1;
 }
 
+/**
+ * An abstract base class for providing arbitrary callbacks for each note
+ * played.
+ */
+export abstract class BasePlayerCallback {
+
+  /* Will be called for each time/note pair in a sequence being played.
+   */
+  abstract run(n: NoteSequence.INote): void;
```

Comment:
should we also pass in time?

Reply:
the reason i'm not passing in time now is that time is now used in the call to `Tone.Draw`. since the whole point of this is to sync arbitrtary events with note playback, it seemed unnecessary to pass in time.
if you'd prefer though, i can obviously pass it in and leave it unused for now.

Reply:
Ack